utils/imgutils/gif.go

```diff
@@ -371,41 +372,20 @@ func (d *decoder) readGraphicControl() error {
 }
 
 func (d *decoder) readImageDescriptor() error {
-	m, err := d.newImageFromDescriptor()
+	err := d.checkImageFromDescriptor()
 	if err != nil {
 		return err
 	}
 	useLocalColorTable := d.imageFields&fColorTable != 0
 	if useLocalColorTable {
-		m.Palette, err = d.readColorTable(d.imageFields)
+		_, err = d.readColorTable(d.imageFields)
```

Comment:
0/5, if we want, we can split `readColorTable` into 2 parts: one which reads, and second which allocates the `color.Palette` slice, and then call just the first part here. This would avoid an additional slice allocation.

Reply:
I think we can simply remove the slice allocation which is totally unnecessary for our use-case. This code shouldn't be used to decode GIFs into raw images.

Reply:
Nice, let's do that